Upgrade to Elasticsearch 2.0.0

@dermidgen I'm referring to your issue in this repo and we'll continue here.

Refs https://github.com/pires/docker-elasticsearch/issues/6

```
io.fabric8.kubernetes.client.KubernetesClientException: Error executing: GET at: https://kubernetes.default.svc/api/v1/namespaces/default/endpoints/elasticsearch. Cause: access denied ("java.net.NetPermission" "getProxySelector")
    at io.fabric8.kubernetes.client.dsl.base.OperationSupport.requestException(OperationSupport.java:245)
    at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:182)
    at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleGet(OperationSupport.java:173)
    at io.fabric8.kubernetes.client.dsl.base.BaseOperation.handleGet(BaseOperation.java:472)
    at io.fabric8.kubernetes.client.dsl.base.BaseOperation.get(BaseOperation.java:105)
    at io.fabric8.elasticsearch.cloud.kubernetes.KubernetesAPIServiceImpl.endpoints(KubernetesAPIServiceImpl.java:27)
    at io.fabric8.elasticsearch.discovery.kubernetes.KubernetesUnicastHostsProvider.buildDynamicNodes(KubernetesUnicastHostsProvider.java:81)
    at org.elasticsearch.discovery.zen.ping.unicast.UnicastZenPing.sendPings(UnicastZenPing.java:316)
    at org.elasticsearch.discovery.zen.ping.unicast.UnicastZenPing$2$1.doRun(UnicastZenPing.java:234)
    at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
    at java.lang.Thread.run(Thread.java:745)
Caused by: java.security.AccessControlException: access denied ("java.net.NetPermission" "getProxySelector")
    at java.security.AccessControlContext.checkPermission(AccessControlContext.java:472)
    at java.security.AccessController.checkPermission(AccessController.java:884)
    at java.lang.SecurityManager.checkPermission(SecurityManager.java:549)
    at java.net.ProxySelector.getDefault(ProxySelector.java:94)
    at com.squareup.okhttp.OkHttpClient.copyWithDefaults(OkHttpClient.java:614)
    at com.squareup.okhttp.Call.<init>(Call.java:48)
    at com.squareup.okhttp.OkHttpClient.newCall(OkHttpClient.java:595)
    at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:180)
```

--- a/README.md
+++ b/README.md
@@ -1,5 +1,5 @@
 # kubernetes-elasticsearch-cluster
-Elasticsearch (1.7.2) cluster on top of Kubernetes made easy.
+Elasticsearch (2.0.0) cluster on top of Kubernetes made easy.
 
 Elasticsearch best-practices recommend to separate nodes in three roles:
 * `Master` nodes - intended for clustering management only, no data, no HTTP API
@@ -12,7 +12,7 @@ Given this, I'm hereby making possible for you to scale as needed. For instance,
 
 ## Pre-requisites
 
-* Kubernetes cluster (tested with v1.0.6 on top of [Vagrant + CoreOS](https://github.com/pires/kubernetes-vagrant-coreos-cluster))
+* Kubernetes cluster (tested with v1.1.1 on top of [Vagrant + CoreOS](https://github.com/pires/kubernetes-vagrant-coreos-cluster))
 * `kubectl` configured to access your cluster master API Server
 
 ## Build images (optional)
@@ -42,34 +42,47 @@ kubectl create -f es-data-rc.yaml
 
 Wait until `es-data` is provisioned.
 
-Now, I leave up to you how to validate the cluster, but a first step is to wait for containers to be in ```RUNNING``` state and check the Elasticsearch master logs:
+Now, I leave up to you how to validate the cluster, but a first step is to wait for containers to be in the `Running` state and check Elasticsearch master logs:
 
 ```
-$ kubectl get pods
+$ kubectl get svc,rc,pods
+NAME                      CLUSTER_IP      EXTERNAL_IP   PORT(S)         SELECTOR                              AGE
+elasticsearch             10.100.13.117                 9200/TCP        component=elasticsearch,role=client   29m
+elasticsearch-discovery   10.100.232.72   <none>        9300/TCP        component=elasticsearch,role=master   29m
+kube-dns                  10.100.0.10     <none>        53/UDP,53/TCP   k8s-app=kube-dns                      50m
+kubernetes                10.100.0.1      <none>        443/TCP         <none>                                53m
+CONTROLLER   CONTAINER(S)   IMAGE(S)                                              SELECTOR                              REPLICAS   AGE
+es-client    es-client      quay.io/pires/docker-elasticsearch-kubernetes:2.0.0   component=elasticsearch,role=client   1          7m
+es-data      es-data        quay.io/pires/docker-elasticsearch-kubernetes:2.0.0   component=elasticsearch,role=data     1          7m
+es-master    es-master      quay.io/pires/docker-elasticsearch-kubernetes:2.0.0   component=elasticsearch,role=master   1          29m
+kube-dns     etcd           gcr.io/google_containers/etcd:2.0.9                   k8s-app=kube-dns                      1          50m
+             kube2sky       gcr.io/google_containers/kube2sky:1.11
+             skydns         gcr.io/google_containers/skydns:2015-03-11-001
 NAME              READY     STATUS    RESTARTS   AGE
-es-client-2ep9o   1/1       Running   0          2m
-es-data-r9tgv     1/1       Running   0          1m
-es-master-vxl6c   1/1       Running   0          6m
+es-client-5lb50   1/1       Running   0          7m
+es-data-0x7e0     1/1       Running   0          7m
+es-master-qsovc   1/1       Running   0          29m
+kube-dns-vkv57    3/3       Running   0          50m
 ```
 
 ```
-$ kubectl logs es-master-vxl6c
+$ kubectl logs es-master-qsovc
 log4j:WARN No such property [maxBackupIndex] in org.apache.log4j.DailyRollingFileAppender.
 log4j:WARN No such property [maxBackupIndex] in org.apache.log4j.DailyRollingFileAppender.
 log4j:WARN No such property [maxBackupIndex] in org.apache.log4j.DailyRollingFileAppender.
-[2015-08-21 10:58:51,324][INFO ][node                     ] [Arc] version[1.7.2], pid[8], build[b88f43f/2015-07-29T09:54:16Z]
-[2015-08-21 10:58:51,328][INFO ][node                     ] [Arc] initializing ...
-[2015-08-21 10:58:51,542][INFO ][plugins                  ] [Arc] loaded [cloud-kubernetes], sites []
-[2015-08-21 10:58:51,624][INFO ][env                      ] [Arc] using [1] data paths, mounts [[/data (/dev/sda9)]], net usable_space [14.4gb], net total_space [15.5gb], types [ext4]
-[2015-08-21 10:58:57,439][INFO ][node                     ] [Arc] initialized
-[2015-08-21 10:58:57,439][INFO ][node                     ] [Arc] starting ...
-[2015-08-21 10:58:57,782][INFO ][transport                ] [Arc] bound_address {inet[/0:0:0:0:0:0:0:0:9300]}, publish_address {inet[/10.244.15.2:9300]}
-[2015-08-21 10:58:57,847][INFO ][discovery                ] [Arc] myesdb/-x16XFUzTCC8xYqWoeEOYQ
-[2015-08-21 10:59:05,167][INFO ][cluster.service          ] [Arc] new_master [Arc][-x16XFUzTCC8xYqWoeEOYQ][es-master-vxl6c][inet[/10.244.15.2:9300]]{data=false, master=true}, reason: zen-disco-join (elected_as_master)
-[2015-08-21 10:59:05,202][INFO ][node                     ] [Arc] started
-[2015-08-21 10:59:05,238][INFO ][gateway                  ] [Arc] recovered [0] indices into cluster_state
-[2015-08-21 11:02:28,797][INFO ][cluster.service          ] [Arc] added {[Gideon][4EfhWSqaTqikbK4tI7bODA][es-data-r9tgv][inet[/10.244.59.4:9300]]{master=false},}, reason: zen-disco-receive(join from node[[Gideon][4EfhWSqaTqikbK4tI7bODA][es-data-r9tgv][inet[/10.244.59.4:9300]]{master=false}])
-[2015-08-21 11:03:16,822][INFO ][cluster.service          ] [Arc] added {[Venomm][tFYxwgqGSpOejHLG4umRqg][es-client-2ep9o][inet[/10.244.53.2:9300]]{data=false, master=false},}, reason: zen-disco-receive(join from node[[Venomm][tFYxwgqGSpOejHLG4umRqg][es-client-2ep9o][inet[/10.244.53.2:9300]]{data=false, master=false}])
+[2015-11-19 20:26:13,193][INFO ][node                     ] [Doctor Spectrum] version[2.0.0], pid[10], build[de54438/2015-10-22T08:09:48Z]
+[2015-11-19 20:26:13,199][INFO ][node                     ] [Doctor Spectrum] initializing ...
+[2015-11-19 20:26:13,548][INFO ][plugins                  ] [Doctor Spectrum] loaded [cloud-kubernetes], sites []
+[2015-11-19 20:26:13,590][INFO ][env                      ] [Doctor Spectrum] using [1] data paths, mounts [[/data (/dev/sda9)]], net usable_space [14.2gb], net total_space [15.5gb], spins? [possibly], types [ext4]
+[2015-11-19 20:26:16,413][INFO ][node                     ] [Doctor Spectrum] initialized
+[2015-11-19 20:26:16,413][INFO ][node                     ] [Doctor Spectrum] starting ...
+[2015-11-19 20:26:16,602][INFO ][transport                ] [Doctor Spectrum] publish_address {10.244.81.2:9300}, bound_addresses {10.244.81.2:9300}
+[2015-11-19 20:26:16,641][INFO ][discovery                ] [Doctor Spectrum] myesdb/nm9NRSuXT7mfJ0YGy-CFFA
+[2015-11-19 20:26:21,391][INFO ][cluster.service          ] [Doctor Spectrum] new_master {Doctor Spectrum}{nm9NRSuXT7mfJ0YGy-CFFA}{10.244.81.2}{10.244.81.2:9300}{data=false, master=true}, reason: zen-disco-join(elected_as_master, [0] joins received)
+[2015-11-19 20:26:21,408][INFO ][node                     ] [Doctor Spectrum] started
+[2015-11-19 20:26:21,443][INFO ][gateway                  ] [Doctor Spectrum] recovered [0] indices into cluster_state
+[2015-11-19 20:48:27,303][INFO ][cluster.service          ] [Doctor Spectrum] added {{Termagaira}{rSv0h7NXRg2_Z1MY7fTvRg}{10.244.63.2}{10.244.63.2:9300}{master=false},}, reason: zen-disco-join(join from node[{Termagaira}{rSv0h7NXRg2_Z1MY7fTvRg}{10.244.63.2}{10.244.63.2:9300}{master=false}])
+[2015-11-19 20:48:31,288][INFO ][cluster.service          ] [Doctor Spectrum] added {{War V}{Rq6n51RlRFyt1VIgG54byQ}{10.244.13.2}{10.244.13.2:9300}{data=false, master=false},}, reason: zen-disco-join(join from node[{War V}{Rq6n51RlRFyt1VIgG54byQ}{10.244.13.2}{10.244.13.2:9300}{data=false, master=false}])
 ```
 
 As you can assert, the cluster is up and running. Easy, wasn't it?
@@ -89,69 +102,69 @@ Did it work?
 ```
 $ kubectl get pods
 NAME              READY     STATUS    RESTARTS   AGE
-es-client-2ep9o   1/1       Running   0          4m
-es-client-ye5s1   1/1       Running   0          50s
-es-data-8az22     1/1       Running   0          47s
-es-data-r9tgv     1/1       Running   0          3m
-es-master-57h7k   1/1       Running   0          52s
-es-master-kuwse   1/1       Running   0          52s
-es-master-vxl6c   1/1       Running   0          8m
+es-client-44h9j   1/1       Running   0          6m
+es-client-5lb50   1/1       Running   0          40m
+es-data-0x7e0     1/1       Running   0          40m
+es-data-jidcs     1/1       Running   0          1m
+es-master-ic3fd   1/1       Running   0          8m
+es-master-qsovc   1/1       Running   0          1h
+es-master-vvuba   1/1       Running   0          8m
+kube-dns-vkv57    3/3       Running   0          1h
 ```
 
 Let's take another look of the Elasticsearch master logs:
 ```
-$ kubectl logs es-master-vxl6c
+$ kubectl logs es-master-qsovc
 log4j:WARN No such property [maxBackupIndex] in org.apache.log4j.DailyRollingFileAppender.
 log4j:WARN No such property [maxBackupIndex] in org.apache.log4j.DailyRollingFileAppender.
 log4j:WARN No such property [maxBackupIndex] in org.apache.log4j.DailyRollingFileAppender.
-[2015-08-21 10:58:51,324][INFO ][node                     ] [Arc] version[1.7.2], pid[8], build[b88f43f/2015-07-29T09:54:16Z]
-[2015-08-21 10:58:51,328][INFO ][node                     ] [Arc] initializing ...
-[2015-08-21 10:58:51,542][INFO ][plugins                  ] [Arc] loaded [cloud-kubernetes], sites []
-[2015-08-21 10:58:51,624][INFO ][env                      ] [Arc] using [1] data paths, mounts [[/data (/dev/sda9)]], net usable_space [14.4gb], net total_space [15.5gb], types [ext4]
-[2015-08-21 10:58:57,439][INFO ][node                     ] [Arc] initialized
-[2015-08-21 10:58:57,439][INFO ][node                     ] [Arc] starting ...
-[2015-08-21 10:58:57,782][INFO ][transport                ] [Arc] bound_address {inet[/0:0:0:0:0:0:0:0:9300]}, publish_address {inet[/10.244.15.2:9300]}
-[2015-08-21 10:58:57,847][INFO ][discovery                ] [Arc] myesdb/-x16XFUzTCC8xYqWoeEOYQ
-[2015-08-21 10:59:05,167][INFO ][cluster.service          ] [Arc] new_master [Arc][-x16XFUzTCC8xYqWoeEOYQ][es-master-vxl6c][inet[/10.244.15.2:9300]]{data=false, master=true}, reason: zen-disco-join (elected_as_master)
-[2015-08-21 10:59:05,202][INFO ][node                     ] [Arc] started
-[2015-08-21 10:59:05,238][INFO ][gateway                  ] [Arc] recovered [0] indices into cluster_state
-[2015-08-21 11:02:28,797][INFO ][cluster.service          ] [Arc] added {[Gideon][4EfhWSqaTqikbK4tI7bODA][es-data-r9tgv][inet[/10.244.59.4:9300]]{master=false},}, reason: zen-disco-receive(join from node[[Gideon][4EfhWSqaTqikbK4tI7bODA][es-data-r9tgv][inet[/10.244.59.4:9300]]{master=false}])
-[2015-08-21 11:03:16,822][INFO ][cluster.service          ] [Arc] added {[Venomm][tFYxwgqGSpOejHLG4umRqg][es-client-2ep9o][inet[/10.244.53.2:9300]]{data=false, master=false},}, reason: zen-disco-receive(join from node[[Venomm][tFYxwgqGSpOejHLG4umRqg][es-client-2ep9o][inet[/10.244.53.2:9300]]{data=false, master=false}])
-[2015-08-21 11:04:40,781][INFO ][cluster.service          ] [Arc] added {[Erik Josten][QUJlahfLTi-MsxzM6_Da0g][es-master-kuwse][inet[/10.244.59.5:9300]]{data=false, master=true},}, reason: zen-disco-receive(join from node[[Erik Josten][QUJlahfLTi-MsxzM6_Da0g][es-master-kuwse][inet[/10.244.59.5:9300]]{data=false, master=true}])
-[2015-08-21 11:04:41,076][INFO ][cluster.service          ] [Arc] added {[Power Princess][V4qnR-6jQOS5ovXQsPgo7g][es-master-57h7k][inet[/10.244.53.3:9300]]{data=false, master=true},}, reason: zen-disco-receive(join from node[[Power Princess][V4qnR-6jQOS5ovXQsPgo7g][es-master-57h7k][inet[/10.244.53.3:9300]]{data=false, master=true}])
-[2015-08-21 11:04:53,966][INFO ][cluster.service          ] [Arc] added {[Cagliostro][Wpfx5fkBRiG2qCEWd8laaQ][es-client-ye5s1][inet[/10.244.15.3:9300]]{data=false, master=false},}, reason: zen-disco-receive(join from node[[Cagliostro][Wpfx5fkBRiG2qCEWd8laaQ][es-client-ye5s1][inet[/10.244.15.3:9300]]{data=false, master=false}])
-[2015-08-21 11:04:56,803][INFO ][cluster.service          ] [Arc] added {[Thog][vkdEtX3ESfWmhXXf-Wi0_Q][es-data-8az22][inet[/10.244.15.4:9300]]{master=false},}, reason: zen-disco-receive(join from node[[Thog][vkdEtX3ESfWmhXXf-Wi0_Q][es-data-8az22][inet[/10.244.15.4:9300]]{master=false}])
+[2015-11-19 20:26:13,193][INFO ][node                     ] [Doctor Spectrum] version[2.0.0], pid[10], build[de54438/2015-10-22T08:09:48Z]
+[2015-11-19 20:26:13,199][INFO ][node                     ] [Doctor Spectrum] initializing ...
+[2015-11-19 20:26:13,548][INFO ][plugins                  ] [Doctor Spectrum] loaded [cloud-kubernetes], sites []
+[2015-11-19 20:26:13,590][INFO ][env                      ] [Doctor Spectrum] using [1] data paths, mounts [[/data (/dev/sda9)]], net usable_space [14.2gb], net total_space [15.5gb], spins? [possibly], types [ext4]
+[2015-11-19 20:26:16,413][INFO ][node                     ] [Doctor Spectrum] initialized
+[2015-11-19 20:26:16,413][INFO ][node                     ] [Doctor Spectrum] starting ...
+[2015-11-19 20:26:16,602][INFO ][transport                ] [Doctor Spectrum] publish_address {10.244.81.2:9300}, bound_addresses {10.244.81.2:9300}
+[2015-11-19 20:26:16,641][INFO ][discovery                ] [Doctor Spectrum] myesdb/nm9NRSuXT7mfJ0YGy-CFFA
+[2015-11-19 20:26:21,391][INFO ][cluster.service          ] [Doctor Spectrum] new_master {Doctor Spectrum}{nm9NRSuXT7mfJ0YGy-CFFA}{10.244.81.2}{10.244.81.2:9300}{data=false, master=true}, reason: zen-disco-join(elected_as_master, [0] joins received)
+[2015-11-19 20:26:21,408][INFO ][node                     ] [Doctor Spectrum] started
+[2015-11-19 20:26:21,443][INFO ][gateway                  ] [Doctor Spectrum] recovered [0] indices into cluster_state
+[2015-11-19 20:48:27,303][INFO ][cluster.service          ] [Doctor Spectrum] added {{Termagaira}{rSv0h7NXRg2_Z1MY7fTvRg}{10.244.63.2}{10.244.63.2:9300}{master=false},}, reason: zen-disco-join(join from node[{Termagaira}{rSv0h7NXRg2_Z1MY7fTvRg}{10.244.63.2}{10.244.63.2:9300}{master=false}])
+[2015-11-19 20:48:31,288][INFO ][cluster.service          ] [Doctor Spectrum] added {{War V}{Rq6n51RlRFyt1VIgG54byQ}{10.244.13.2}{10.244.13.2:9300}{data=false, master=false},}, reason: zen-disco-join(join from node[{War V}{Rq6n51RlRFyt1VIgG54byQ}{10.244.13.2}{10.244.13.2:9300}{data=false, master=false}])
+[2015-11-19 21:17:54,140][INFO ][cluster.service          ] [Doctor Spectrum] added {{Crossbones}{IVfzPHf8QkS8EX68pEv9oQ}{10.244.63.3}{10.244.63.3:9300}{data=false, master=true},}, reason: zen-disco-join(join from node[{Crossbones}{IVfzPHf8QkS8EX68pEv9oQ}{10.244.63.3}{10.244.63.3:9300}{data=false, master=true}])
+[2015-11-19 21:17:54,281][INFO ][cluster.service          ] [Doctor Spectrum] added {{Lady Lark}{TSoI37adQPaNpEDJ2c9bsw}{10.244.13.3}{10.244.13.3:9300}{data=false, master=true},}, reason: zen-disco-join(join from node[{Lady Lark}{TSoI37adQPaNpEDJ2c9bsw}{10.244.13.3}{10.244.13.3:9300}{data=false, master=true}])
+[2015-11-19 21:20:33,233][INFO ][cluster.service          ] [Doctor Spectrum] added {{Ben Urich}{P0EqCsVGRhOI5IRAWH34yg}{10.244.81.3}{10.244.81.3:9300}{data=false, master=false},}, reason: zen-disco-join(join from node[{Ben Urich}{P0EqCsVGRhOI5IRAWH34yg}{10.244.81.3}{10.244.81.3:9300}{data=false, master=false}])
+[2015-11-19 21:25:41,480][INFO ][cluster.service          ] [Doctor Spectrum] added {{Cassandra Nova}{DeElUqr2QdyifWwJU68hcw}{10.244.13.4}{10.244.13.4:9300}{master=false},}, reason: zen-disco-join(join from node[{Cassandra Nova}{DeElUqr2QdyifWwJU68hcw}{10.244.13.4}{10.244.13.4:9300}{master=false}])
 ```
 
 ### Access the service
 
-*Don't forget* that services in Kubernetes are only acessible from containers in the cluster. For different behavior you should [configure the creation of an external load-balancer](http://kubernetes.io/v1.0/docs/user-guide/services.html#type-loadbalancer). While it's supported within this example service descriptor, its usage is out of scope of this document, for now.
+*Don't forget* that services in Kubernetes are only acessible from containers in the cluster. For different behavior you should [configure the creation of an external load-balancer](http://kubernetes.io/v1.1/docs/user-guide/services.html#type-loadbalancer). While it's supported within this example service descriptor, its usage is out of scope of this document, for now.
 
 ```
 $ kubectl get service elasticsearch
-NAME            LABELS                                SELECTOR                              IP(S)          PORT(S)
-elasticsearch   component=elasticsearch,role=client   component=elasticsearch,role=client   10.100.134.2   9200/TCP
+NAME            CLUSTER_IP      EXTERNAL_IP   PORT(S)    SELECTOR                              AGE
+elasticsearch   10.100.13.117                 9200/TCP   component=elasticsearch,role=client   1h
 ```
 
 From any host on your cluster (that's running `kube-proxy`), run:
 
 ```
-curl http://10.100.134.2:9200
+curl http://10.100.13.117:9200
 ```
 
 You should see something similar to the following:
 
 ```json
 {
-  "status" : 200,
-  "name" : "Cagliostro",
+  "name" : "War V",
   "cluster_name" : "myesdb",
   "version" : {
-    "number" : "1.7.2",
-    "build_hash" : "b88f43fc40b0bcd7f173a1f9ee2e97816de80b19",
-    "build_timestamp" : "2015-07-29T09:54:16Z",
+    "number" : "2.0.0",
+    "build_hash" : "de54438d6af8f9340d50c5c786151783ce7d6be5",
+    "build_timestamp" : "2015-10-22T08:09:48Z",
     "build_snapshot" : false,
-    "lucene_version" : "4.10.4"
+    "lucene_version" : "5.2.1"
   },
   "tagline" : "You Know, for Search"
 }
@@ -160,7 +173,7 @@ You should see something similar to the following:
 Or if you want to see cluster information:
 
 ```
-curl http://10.100.134.2:9200/_cluster/health?pretty
+curl http://10.100.13.117:9200/_cluster/health?pretty
 ```
 
 You should see something similar to the following:
@@ -179,6 +192,8 @@ You should see something similar to the following:
   "unassigned_shards" : 0,
   "delayed_unassigned_shards" : 0,
   "number_of_pending_tasks" : 0,
-  "number_of_in_flight_fetch" : 0
+  "number_of_in_flight_fetch" : 0,
+  "task_max_waiting_in_queue_millis" : 0,
+  "active_shards_percent_as_number" : 100.0
 }
 ```

--- a/es-client-rc.yaml
+++ b/es-client-rc.yaml
@@ -20,10 +20,11 @@ spec:
       containers:
       - name: es-client
         securityContext:
+          privileged: true
           capabilities:
             add:
               - IPC_LOCK
-        image: quay.io/pires/docker-elasticsearch-kubernetes:1.7.2
+        image: quay.io/pires/docker-elasticsearch-kubernetes:2.0.0
         env:
         - name: KUBERNETES_CA_CERTIFICATE_FILE
           value: /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
@@ -50,5 +51,6 @@ spec:
         - mountPath: /data
           name: storage
       volumes:
-      - name: storage
-        emptyDir: {}
+          - emptyDir:
+              medium: ""
+            name: "storage"

--- a/es-data-rc.yaml
+++ b/es-data-rc.yaml
@@ -20,10 +20,11 @@ spec:
       containers:
       - name: es-data
         securityContext:
+          privileged: true
           capabilities:
             add:
               - IPC_LOCK
-        image: quay.io/pires/docker-elasticsearch-kubernetes:1.7.2
+        image: quay.io/pires/docker-elasticsearch-kubernetes:2.0.0
         env:
         - name: KUBERNETES_CA_CERTIFICATE_FILE
           value: /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
@@ -45,5 +46,6 @@ spec:
         - mountPath: /data
           name: storage
       volumes:
-      - name: storage
-        emptyDir: {}
+          - emptyDir:
+              medium: ""
+            name: "storage"

--- a/es-master-rc.yaml
+++ b/es-master-rc.yaml
@@ -20,10 +20,11 @@ spec:
       containers:
       - name: es-master
         securityContext:
+          privileged: true
           capabilities:
             add:
               - IPC_LOCK
-        image: quay.io/pires/docker-elasticsearch-kubernetes:1.7.2
+        image: quay.io/pires/docker-elasticsearch-kubernetes:2.0.0
         env:
         - name: KUBERNETES_CA_CERTIFICATE_FILE
           value: /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
@@ -47,5 +48,6 @@ spec:
         - mountPath: /data
           name: storage
       volumes:
-      - name: storage
-        emptyDir: {}
+          - emptyDir:
+              medium: ""
+            name: "storage"

